plugins/ssrmatching/ssrmatching.ml

```diff
@@ -336,7 +336,7 @@ let proj_nparams c =
   with Not_found -> 0
 
 let isRigid c = match kind c with
-  | Prod _ | Sort _ | Lambda _ | Case _ | Fix _ | CoFix _ -> true
+  | Prod _ | Sort _ | Lambda _ | Case _ | Fix _ | CoFix _ | Int _ | Float _ | Array _ -> true
   | _ -> false
```

Comment:
You may want to take the opportunity to remove that catch all pattern, so that the bug doesn't happen again if another primitive type get added in the future.

Reply:
This is a good advice. @Juancoro if you don't mind following it... Otherwise, if you don't have time, I can merge it as is and leave it as a further improvement.

Reply:
Sure! I'll do it right away. Thanks for the advice!

Reply:
Done!

Reply:
Thanks!
(not sure it is worth protecting against a change of arity of some constructors (`Evar (_, _)` instead of `Evar _` for instance), but why not)

Reply:
If you think it's better to have a wildcard instead of the arity I've no problem in changing it. Just tell me to which constructors does this apply (or rather, if it makes sense to have the arity at all) and I'll happily change it!

Reply:
It just got merged! Never mind then :smile: 
Thanks to the both of you for your suggestions!

Reply:
It's fine, I was just putting this side note in case you'd like to save a few characters next time ;-)